### PR TITLE
Farm V4 Onboarding Modal And User Cache FIxes

### PR DESCRIPTION
**Pull request by deadeagle63**
Increased stability and guaranteed userCache exists to prevent the bunch of optional accessors

## Types of changes

- [x] Technical Debt (non-breaking change which removes unused code/assets)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] The related ClickUp task has been linked to this PR
- [x] The person creating the pull request is listed in "Assignees"
- [ ] Change requires updated documentation

### CODE REVIEW

File: src/pages/FarmV4/FarmContainer.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const FarmContainer: FC = () => {
     setPool
   } = useFarmContext()
   const [searchTokens, setSearchTokens] = useState<string>('')
-  const [showDeposited, setShowDeposited] = useState<boolean>(existingUserCache.farm.showDepositedFilter)
+  const [showDeposited, setShowDeposited] = useState<boolean>(userCache.farm.showDepositedFilter)
```
Comment (Shashank729) [inline src/pages/FarmV4/FarmContainer.tsx]: This should be:

existingUserCache.gamma.showDeposited filter.

We don't want to update Farm filter here.

It was wrong earlier too.

File: src/pages/FarmV4/FarmContainer.tsx [hunk 2]
```diff
@@ -75,18 +75,17 @@ export const FarmContainer: FC = () => {
...
-          'gfx-user-cache',
-          JSON.stringify({
-            ...existingUserCache,
-            farm: { ...existingUserCache.farm, showDepositedFilter: false }
-          })
-        )
+        updateUserCache({
+          farm: {
+            ...userCache.farm,
+            showDepositedFilter: false
```
Comment (Shashank729) [inline src/pages/FarmV4/FarmContainer.tsx:82]: Same here

File: src/pages/FarmV4/FarmContainer.tsx [hunk 3]
```diff
@@ -99,13 +98,13 @@ export const FarmContainer: FC = () => {
...
-        'gfx-user-cache',
-        JSON.stringify({
-          ...existingUserCache,
-          farm: { ...existingUserCache.farm, showDepositedFilter: !prev }
-        })
-      )
+      updateUserCache({
+        farm: {
+          ...userCache.farm,
+          showDepositedFilter: !prev
```
Comment (Shashank729) [inline src/pages/FarmV4/FarmContainer.tsx:103]: Same here

File: src/pages/FarmV4/FarmContainer.tsx (hunk 1 above)
Comment (deadeagle63) [reply src/pages/FarmV4/FarmContainer.tsx]: It is not, it is userCache - if you scroll up you will see it imports it from the hook

Comment (deadeagle63) [reply src/pages/FarmV4/FarmContainer.tsx]: The code previously updated this, was the original code wrong then?

Comment (Shashank729) [reply src/pages/FarmV4/FarmContainer.tsx]: Yes, it was wrong. It is the showDeposited filter for Farm (and not ssl). So we just need to update the filter for 'gamma' case & not 'farm' case. It already exists in farmv3. Farmv4 = Gamma related changes.

**Comment by walt-1** [comment]
i pushed the fix

File: src/pages/FarmV4/FarmContainer.tsx (hunk 1 above)
Comment (deadeagle63) [reply src/pages/FarmV4/FarmContainer.tsx]: cool luke fixed it

**Review by walt-1 — APPROVED** [review]

**Pull request merged**